src/Test.UnitTests.Sarif.Converters/FortifyFprConverterTests.cs

```diff
@@ -94,7 +94,58 @@ public void FortifyFprConverter_GetFailureLevelFromRuleMetadata_ReturnsAppropria
 
                 level.Should().Be(keyValuePair.Value);
             }
-            
         }
+
+        [Fact]
+        public void FortifyFprConverter_GetOriginalUriBaseIdsDictionary_sourceIsDriveLetter()
+        {
+            Dictionary<string, ArtifactLocation> originalUriBaseIdsDictionary = FortifyFprConverter.GetOriginalUriBaseIdsDictionary("C:", "Windows Server 2016");
+
+            originalUriBaseIdsDictionary.Count.Should().Be(1);
+            originalUriBaseIdsDictionary.ContainsKey(FortifyFprConverter.FileLocationUriBaseId).Should().BeTrue();
+
+            originalUriBaseIdsDictionary[FortifyFprConverter.FileLocationUriBaseId].Uri.Should().Be(@"file:///C:/");
+            originalUriBaseIdsDictionary[FortifyFprConverter.FileLocationUriBaseId].UriBaseId.Should().BeNull();
+
+        }
+
+        [Fact]
+        public void FortifyFprConverter_GetOriginalUriBaseIdsDictionary_sourceIsAbsolutePathWithoutTrailingSlash()
+        {
+            Dictionary<string, ArtifactLocation> originalUriBaseIdsDictionary = FortifyFprConverter.GetOriginalUriBaseIdsDictionary("C:/test/123", "Windows 10");
+
+            originalUriBaseIdsDictionary.Count.Should().Be(1);
+            originalUriBaseIdsDictionary.ContainsKey(FortifyFprConverter.FileLocationUriBaseId).Should().BeTrue();
+            originalUriBaseIdsDictionary[FortifyFprConverter.FileLocationUriBaseId].Uri.Should().Be(@"file:///C:/test/123/");
+            originalUriBaseIdsDictionary[FortifyFprConverter.FileLocationUriBaseId].UriBaseId.Should().BeNull();
+        }
+
+        [Fact]
+        public void FortifyFprConverter_GetOriginalUriBaseIdsDictionary_sourceIsLinuxStyleAbsolutePath()
+        {
+            Dictionary<string, ArtifactLocation> originalUriBaseIdsDictionary = FortifyFprConverter.GetOriginalUriBaseIdsDictionary("/root/projects/myproject/src/", "Linux");
+
+            originalUriBaseIdsDictionary.Count.Should().Be(1);
+            originalUriBaseIdsDictionary.ContainsKey(FortifyFprConverter.FileLocationUriBaseId).Should().BeTrue();
+            originalUriBaseIdsDictionary[FortifyFprConverter.FileLocationUriBaseId].Uri.Should().Be(@"file://root/projects/myproject/src/");
```

Comment:
>// [](start = 107, length = 2)

Existing bug: this two-slash form isn't right. See [https://tools.ietf.org/html/rfc8089#appendix-B](https://tools.ietf.org/html/rfc8089#appendix-B) (RFC 8089 : The "file" URI Scheme). The URI for a local file is either
```
[file:///path/to/file](file:///path/to/file)
```
or
```
file:/path/to/file
```
With two slashes, the first component is interpreted as the "authority":
```
[file://host.example.com/path/to/file](file://host.example.com/path/to/file)
```
In your example, `[file://root/projects/myproject/src/`](file://root/projects/myproject/src/`) means "the file `/projects/myproject/src/` on host `root`.

Reply:
thanks! fixed!

---
In reply to: [298217975](https://github.com/microsoft/sarif-sdk/pull/1547#discussion_r298217975) [](ancestors = 298217975)